tests/test_ag3.py

```diff
@@ -576,3 +576,66 @@ def test_snp_effects():
     assert df.loc[646].effect == 'SPLICE_CORE'
     assert df.loc[652].effect == 'SPLICE_REGION'
     assert df.loc[674].effect == 'INTRONIC'
+
+
+def test_snp_allele_frequencies():
+    ag3 = setup_ag3()
+    populations = {
+        "ke": "country == 'Kenya'",
+        "bf_2012_col": "country == 'Burkina Faso' and year == 2012 and species == 'coluzzii'",
+    }
+    expected_fields = ["position",
+                       "ref_allele",
+                       "alt_allele",
+                       "ke",
+                       "bf_2012_col",
+                       "maximum",
+                       ]
+    # drop invariants
+    df = ag3.snp_allele_frequencies(transcript="AGAP009194-RA",
+                                    populations=populations,
+                                    site_mask='gamb_colu',
+                                    sample_sets="v3_wild",
+                                    drop_invariants=True)
+
+    assert isinstance(df, pandas.DataFrame)
+    assert expected_fields == df.columns.tolist()
+    assert df.shape == (133, 6)
+    assert df.loc[4].position == 28597653
+    assert df.loc[5].ref_allele == "A"
+    assert df.loc[13].alt_allele == "C"
+    assert df.loc[16].ke == 0
+    assert df.loc[22].bf_2012_col == pytest.approx(0.006097, abs=1e-6)
+    assert df.loc[39].maximum == pytest.approx(0.006097, abs=1e-6)
+    #check invariants have been dropped
+    assert df.maximum.min() > 0
+
+    populations = {
+        "gm": "country == 'Gambia, The'",
+        "mz": "country == 'Mozambique' and year == 2004"
+    }
+    expected_fields = ["position",
+                       "ref_allele",
+                       "alt_allele",
+                       "gm",
+                       "mz",
+                       "maximum",
+                       ]
+    # keep invariants
+    df = ag3.snp_allele_frequencies(transcript="AGAP004707-RD",
+                                    populations=populations,
+                                    site_mask='gamb_colu',
+                                    sample_sets="v3_wild",
+                                    drop_invariants=False)
+
+    assert isinstance(df, pandas.DataFrame)
+    assert expected_fields == df.columns.tolist()
+    assert df.shape == (132306, 6)
+    assert df.loc[0].position == 2358158
+    assert df.loc[1].ref_allele == "A"
+    assert df.loc[2].alt_allele == "G"
+    assert df.loc[3].gm == 0.0
+    assert df.loc[4].mz == 0.0
+    assert df.loc[72].maximum == pytest.approx(0.001792, abs=1e-6)
+    # check invariant positions are still present
+    assert 0 in df.maximum.unique()
```

Comment:
```suggestion
    assert np.any(df.maximum == 0)
```

Slightly clearer perhaps.